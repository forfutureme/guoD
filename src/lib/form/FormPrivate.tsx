/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-15 07:49:24
 * @Desc: 表单单纯版
 */
import React from 'react'

import Input from '@lib/input/Input'
import './FormPrivate.scss'
import {joinClassName, prefixClass} from '@utils/classNameHandle'
const fixClass = prefixClass('guo-form')
export type formDataProps = {
  [key: string]: any
}

interface props extends React.FormHTMLAttributes<HTMLFormElement> {
  formData: formDataProps
  fileds: Array<{name: string; label: string; input: {type: string}}>
  onSubmit: (info: formDataProps) => void
  button: React.ReactFragment
}

const FormPrivate: React.FC<props> = ({
  className,
  formData,
  fileds,
  onSubmit,
  button
}) => {
  const inputChange = (name: string, value: string) => {
    onSubmit({
      [name]: value
    })
  }
  const submit = (e: React.FormEvent) => {
    e.preventDefault()
    onSubmit(formData)
  }
  return (
    <form
      className={joinClassName(fixClass('private'), className)}
      onSubmit={submit}>
      {fileds.map(({name, label, input}) => (
        <div key={name} className="box">
          <label className="label">{label}</label>
          <Input
            type={input.type}
            value={formData[name]}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
              inputChange(name, e.target.value)
            }}
          />
        </div>
      ))}
      <div>{button}</div>
    </form>
  )
}

export default FormPrivate
