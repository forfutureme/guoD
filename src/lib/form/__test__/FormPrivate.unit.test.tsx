/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-15 10:41:46
 * @Desc: FormPrivate 单元测试
 */
import renderer from 'react-test-renderer'
import React from 'react'
import FormPrivate from '../FormPrivate'

describe('formPrivate测试', () => {
  it('测试formPrivate渲染', () => {
    const formData = {
      name: ''
    }
    const fileds = [
      {
        label: '姓名',
        name: 'name',
        input: {type: 'text'}
      }
    ]
    const json = renderer
      .create(
        <FormPrivate
          formData={formData}
          fileds={fileds}
          onSubmit={() => {}}
          button={<></>}
        />
      )
      .toJSON()
    expect(json).toMatchSnapshot()
  })
})
