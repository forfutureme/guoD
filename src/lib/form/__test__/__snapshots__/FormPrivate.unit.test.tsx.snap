// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formPrivate测试 测试formPrivate渲染 1`] = `
<form
  className="guo-form-private"
  onSubmit={[Function]}
>
  <div
    className="box"
  >
    <label
      className="label"
    >
      姓名
    </label>
    <div
      className="guo-input-box"
    >
      <input
        className="guo-input"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div />
</form>
`;
